examples/gitops/argocd-multi-cluster/hub-cluster/main.tf

```diff
@@ -0,0 +1,372 @@
+  }
+
+  depends_on = [module.eks_blueprints_kubernetes_addons]
+}
+
+
+################################################################################
+# EKS Workloads via ArgoCD
+################################################################################
+
+module "eks_blueprints_argocd_workloads" {
+  source = "github.com/csantanapr/terraform-aws-eks-blueprints-addons//modules/argocd?ref=argo-multi-cluster" #TODO change git org to aws-ia
+
+  argocd_skip_install = true # Skip argocd controller install
+
+  helm_config = {
+    namespace        = local.argocd_namespace
+    create_namespace = false
+  }
+
+  applications = {
+    # This shows how to deploy an application to leverage cluster generator  https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Generators-Cluster/
+    application-set = {
+      add_on_application = false
+      path               = "application-sets"
+      repo_url           = "https://github.com/csantanapr/eks-blueprints-workloads.git"
+      target_revision    = "argo-multi-cluster" #TODO change to main
+      #repo_url             = "git@github.com:csantanapr-test-gitops-1/eks-blueprints-workloads.git" #TODO change to https://github.com/aws-samples/eks-blueprints-workloads once git repo is updated
+      #ssh_key_secret_name  = "github-ssh-key"# Needed for private repos
+      #git_secret_namespace = "argocd"
+      #git_secret_name      = "${local.name}-workloads"
+    }
+
+  }
+
+  addon_context = {
+    aws_region_name                = local.region
+    aws_caller_identity_account_id = data.aws_caller_identity.current.account_id
+    eks_cluster_id                 = module.eks.cluster_name
+  }
+
+  depends_on = [module.eks_blueprints_argocd_addons]
+
+}
+
+################################################################################
+# ArgoCD EKS Access
+################################################################################
+
+module "argocd_irsa" {
+  source = "github.com/csantanapr/terraform-aws-eks-blueprints-addons//modules/eks-blueprints-addon?ref=argo-multi-cluster" #TODO change git org to aws-ia
+
+  create_release             = false
+  create_role                = true
+  role_name_use_prefix       = false
+  role_name                  = "${module.eks.cluster_name}-argocd-hub"
+  assume_role_condition_test = "StringLike"
+  role_policy_arns = {
+    ArgoCD_EKS_Policy = aws_iam_policy.irsa_policy.arn
+  }
+  oidc_providers = {
+    this = {
+      provider_arn    = module.eks.oidc_provider_arn
+      namespace       = local.argocd_namespace
+      service_account = "argocd-*"
+    }
+  }
+  tags = local.tags
+
+}
+
+resource "aws_iam_policy" "irsa_policy" {
+  name        = "${module.eks.cluster_name}-argocd-irsa"
+  description = "IAM Policy for ArgoCD Hub"
+  policy      = data.aws_iam_policy_document.irsa_policy.json
+  tags        = local.tags
+}
+
+data "aws_iam_policy_document" "irsa_policy" {
+  statement {
+    effect    = "Allow"
+    resources = ["*"]
```

Comment:
who or what is assuming? we should have this locked down and not allow anyone anywhere to assume

Reply:
@bryantbiggs

This for the role in argocd-controller and argocd-server in IRSA to assume, What is the syntax to lock it doing to a role?

Reply:
```hcl
data "aws_iam_policy_document" "this" {
  statement {
    effect    = "Allow"
    actions   = ["sts:AssumeRole"]
    principals {
      type        = "AWS"
      identifiers = ["<ROLE-ARN>"]
    }
  }
} 
```

Reply:
I'm not sure if this policy is required. The role in argocd controller will be assuming multiple roles for each spoke cluster register, but this allow to assume is done on the spoke template terraform.